src/anchormanager.rs

```diff
@@ -1,23 +1,69 @@
 // This file is Copyright its original authors, visible in version control
-// // history.
-// //
-// // This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
-// // or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
-// // <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
-// // You may not use this file except in accordance with one or both of these
-// // licenses.
+// history.
 //
-// //! The top-level component of a Civ Kit node responsible to sanitize and
-// //! order trade kinds, counter-sign and anchor them and dispatch them to
-// //! clients according to requests.
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+//
+// This module will receive a serialized payload from the BoardManager and
+// pack them into an anchor(?) and at periodic intervals, counter-sign them
+// with the Board pubkey and submit for validation on the Mainstay server.
+// 
+
+use std::sync::Arc;
+use std::sync::Mutex;
+use bitcoin::secp256k1::{PublicKey};
+
+// use civkit::events;
+// use civkit::events::ClientEvents;
+// use civkit::kindprocessor::KindProcessor;
+
+pub struct AnchorManager {
+    our_service_pubkey: Arc<PublicKey>, 
+    our_service_offers: Arc<SerializedOffers>,     
+}
+
+pub struct SerializedOffers {
```

Comment:
Can we call this `SerializedEvent` as the granular data structure we’re aiming to commit for attestations is more-or-less a Nostr event with some additional metadatas ?